### PR TITLE
test: node back online

### PR DESCRIPTION
**Pull request by ppca**
I'm storing the key share to a local file instead so that when the node restarts in our integration test, it can find its key share.

test fails 50% of the time. Sometimes it is caused by needing to generate presignature when node restarts, and signature request picked old presignatures that the node does not have, then change to the next old presignature, then eventually timed out on the contract sign call. Sometimes other stuff. I'm still looking into how to fix those.

### CODE REVIEW

**Review by volovyks — COMMENTED** [review]
If possible, let's follow this pattern for online/offline node mod changes:
```rust
assert!(ctx.add_participant().await.is_ok());
assert!(ctx.remove_participant(None).await.is_ok());
```
We will need to build tests where we randomly add/remove nodes, put them offline, signing payloads, etc.

File: integration-tests/tests/multichain/mod.rs [hunk 1]
```diff
@@ -308,3 +308,46 @@ async fn test_latest_block_height() -> anyhow::Result<()> {
...
+            // Try again if the first attempt failed. This second portion should not be needed when the NEP
+            // comes in for resumeable MPC.
+            if presig_res.is_err() || sig_res.is_err() {
+                // Retry if the first attempt failed.
+                wait_for::has_at_least_presignatures(&ctx, 2).await?;
+                actions::single_signature_production(&ctx, &state_0).await?;
+            }
+
+            // Start the killed node again
+            ctx.nodes.restart_node(killed_node_config).await?;
```
[the remark this answers is not in the document]
Comment (ppca) [reply integration-tests/tests/multichain/mod.rs:342]: as I mentioned, run() and restart() are different.

[the remark this answers is not in the document]
Comment (volovyks) [reply integration-tests/tests/multichain/mod.rs:342]: Just saying that restart usually means "stop" and then "start". Here it means "start with these parameters".

**Comment by ppca** [comment]
The code will store secret key share locally in disk, and after running each test, it will clean up the key share files.

**Review by ChaoticTempest — DISMISSED** [review]

**Review by ChaoticTempest — APPROVED** [review]

**Pull request merged**